lib/utils/jsonPathUtils.ts

```diff
@@ -58,4 +59,48 @@ export class JsonPathUtils {
     }
     return result;
   }
+
+  public static changePropertyNameRecursively(
+    obj: IInternalPresentationDefinition,
+    currentPropertyName: string,
+    newPropertyName: string
+  ): any {
+    console.log(newPropertyName);
+    const existingPaths: { value: unknown; path: (string | number)[] }[] = JsonPathUtils.extractInputField(obj, [
+      '$..' + currentPropertyName,
+    ]);
+    const newPd = { ...obj };
```

Comment:
This means the objects in the spread are shallow copied (the original objects), meaning you are manipulating the original object as well, which probably is not intended, given you are also returning the property

Reply:
changed to update the instance